### PR TITLE
feat(connector): apple

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

ref: https://developer.apple.com/documentation/sign_in_with_apple

- implement apple universal connector
- implement `useAppleAuth` hook
- directly use AppleID auth SDK in `ui` since it can invoke the native sign in sheet in Apple operating systems

## Linear Issue Reference

LOG-1608

## Testing

Note apple disallow localhost Redirect URI, so deployment is needed for native app testing. (Rewriting `/etc/host` to test in macOS)

image

### Apple operating systems
- [x] sign in with Safari (macOS, native popup)
  - known issue: native popup just appear once, and this attempt will fail for sure
- [x] sign in with Chrome (macOS, window popup)
- [ ] sign in with native app (iOS, native popup)

### Android
- [ ] sign in with browser
- [ ] sign in with native app

### CODE REVIEW

File: packages/console/src/pages/ConnectorDetails/index.tsx
Comment (gao-sun) [inline packages/console/src/pages/ConnectorDetails/index.tsx:51]: @xiaoyijun is this by design? i think redundant content will be filled

image

**Review by darcyYe — COMMENTED** [review]
LGTM, but not familiar with UI/console part, waiting for another approval.

File: packages/ui/src/hooks/use-social.ts
Comment (simeng-li) [inline packages/ui/src/hooks/use-social.ts:19]: Since we have all connectors loaded in the context, one alteration is query out the connector target using the connectorId, so no need to pass in the target any more. Not sure which one is better.

File: packages/ui/src/utils/index.ts
Comment (simeng-li) [inline packages/ui/src/utils/index.ts:38]: is it being used any where?  Didn't find the ref

File: packages/core/src/connectors/index.test.ts
Comment (simeng-li) [inline packages/core/src/connectors/index.test.ts:42]: @wangsijie @darcyYe like mentioned offline, maybe we could clean up these mocks to the __mock__ files as part of the refactor job.

File: packages/console/src/pages/ConnectorDetails/index.tsx
Comment (xiaoyijun) [reply packages/console/src/pages/ConnectorDetails/index.tsx:51]: This aligns with other connectors and needs further discussion.

File: packages/connector-apple/src/index.test.ts
Comment (xiaoyijun) [inline packages/connector-apple/src/index.test.ts:59]: Should we fulfill this part?

File: packages/connector-apple/src/index.ts
Comment (xiaoyijun) [inline packages/connector-apple/src/index.ts:38]: Since this `scope` is empty in the constant file, do we need it here?

File: packages/ui/src/hooks/use-apple-auth.ts
Comment (xiaoyijun) [inline packages/ui/src/hooks/use-apple-auth.ts:10]: Do we need to `removeEventListener` after the script is loaded or failed?

File: packages/console/src/pages/ConnectorDetails/index.tsx
Comment (xiaoyijun) [reply packages/console/src/pages/ConnectorDetails/index.tsx:51]: cc: @wangsijie @darcyYe

File: packages/ui/src/hooks/use-apple-auth.ts
Comment (simeng-li) [inline packages/ui/src/hooks/use-apple-auth.ts:54]: nit: have a util method `parseQueryParameters`

File: packages/console/src/pages/ConnectorDetails/index.tsx
Comment (gao-sun) [reply packages/console/src/pages/ConnectorDetails/index.tsx:51]: @demonzoo

File: packages/ui/src/App.tsx
Comment (gao-sun) [reply packages/ui/src/App.tsx:41]: yes, forgot that

File: packages/ui/src/hooks/use-social.ts
Comment (gao-sun) [reply packages/ui/src/hooks/use-social.ts:19]: actually i would like to pass the whole connector object as the param since it's just an address

File: packages/connector-apple/src/index.test.ts
Comment (gao-sun) [reply packages/connector-apple/src/index.test.ts:59]: was tired yesterday. would you like to help me to do this

File: packages/connector-apple/src/index.ts
Comment (gao-sun) [reply packages/connector-apple/src/index.ts:38]: don't know if it's required, also we can keep this param to remind us to add more scopes in the future if needed

Comment (gao-sun) [reply packages/connector-apple/src/index.ts:38]: don't know if it's required, also we can keep this param to remind us to add more scopes in the future if needed

Comment (gao-sun) [reply packages/connector-apple/src/index.ts:38]: im ok to remove

File: packages/ui/src/hooks/use-apple-auth.ts
Comment (gao-sun) [reply packages/ui/src/hooks/use-apple-auth.ts:10]: i think it's ok, it'll trigger just once

File: packages/ui/src/hooks/use-apple-auth.ts
Comment (gao-sun) [reply packages/ui/src/hooks/use-apple-auth.ts:54]: didn't notice this. will update soon

File: packages/ui/src/App.tsx
Comment (gao-sun) [reply packages/ui/src/App.tsx:41]: discussed offline

File: packages/ui/src/hooks/use-social.ts
Comment (gao-sun) [reply packages/ui/src/hooks/use-social.ts:19]: LOG-2725

File: packages/connector-apple/src/index.test.ts
Comment (xiaoyijun) [reply packages/connector-apple/src/index.test.ts:59]: Add an issue for this one. LOG-2726
I'll do it after our connector is refactored.

**Review by xiaoyijun — APPROVED** [review]
LGTM

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**
